tensorrt_llm/_torch/models/modeling_llama.py

```diff
@@ -652,32 +658,50 @@ def __init__(self, model_config: ModelConfig[LlamaConfig]) -> None:
                                             dtype=torch.int64),
                                 requires_grad=False)
 
+        if self.hidden_size_in != config.hidden_size:
+            self.embed_tokens = Embedding(
+                config.vocab_size,
+                config.hidden_size,
+                dtype=config.torch_dtype,
+                parallel_config=ParallelConfig(
+                    tensor_parallel_rank=model_config.mapping.tp_rank,
+                    tensor_parallel_size=model_config.mapping.tp_size,
+                    tensor_parallel_mode=TensorParallelMode.COLUMN,
+                    pipeline_parallel_size=model_config.mapping.pp_size,
+                    parallel_rank=model_config.mapping.rank,
+                    gather_output=True,
+                    gpus_per_node=model_config.mapping.gpus_per_node,
+                ),
+            )
+        else:
+            # Shared with target model.
+            self.embed_tokens = None
+
     def forward(
         self,
         attn_metadata: AttentionMetadata,
-        embed_tokens: torch.nn.Module,
         input_ids: Optional[torch.LongTensor] = None,
         position_ids: Optional[torch.LongTensor] = None,
         inputs_embeds: Optional[torch.FloatTensor] = None,
         spec_metadata: Optional[SpecMetadata] = None,
         hidden_states: Optional[torch.Tensor] = None,
```

Comment:
The type is incorrect. It's not optional. Is it a single tensor now?

Reply:
Yeah. We have an assertion that it's not none. It has to have this annotation to match the base class unfortunately.